### PR TITLE
Added a component connecting sentiment analysis

### PR DESCRIPTION
**Pull request by thePetrMarek**
The first version of component connecting sentiment analysis.
The url for Triton inference server is currently defined in `app.local.properties`.

### CODE REVIEW

File: lib/src/main/kotlin/org/promethist/core/Input.kt [hunk 1]
```diff
@@ -58,6 +58,12 @@ data class Input(
     @get:JsonIgnore
     val intent get() = intents.firstOrNull() ?: error("No intent class recognized in input")
 
+    @get:JsonIgnore
+    val sentimets get() = classes.filter { it.type == Class.Type.Sentiment }
+
+    @get:JsonIgnore
+    val sentiment get() = classes.firstOrNull() ?: error("No sentiment class recognized in input")
```
Comment (xzayda) [inline lib/src/main/kotlin/org/promethist/core/Input.kt:65]: I prefer to return ?: Class(Class.Type.Sentiment, Sentiment.UNKNOWN.name) to ensure that sentiment sensitive code will work even when sentiment detection not available

File: runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    )
+
+    override fun process(context: Context): Context {
+
+        logger.info("Calling Sentiment with input ${context.input}")
+
+        val text = context.input.transcript.text
+        val input = Input(listOf(text))
+        val request = Request(listOf(input))
+        val response = webTarget.path("/sentiment/infer").request().post(Entity.json(request), object : GenericType<Response>() {})
```
Comment (xzayda) [inline runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt:34]: add try ... catch to make it work event when embedded service not available (take a look on Cassandra class implementation)

File: lib/src/main/kotlin/org/promethist/core/Input.kt [hunk 3]
```diff
@@ -58,6 +58,12 @@ data class Input(
     @get:JsonIgnore
     val intent get() = intents.firstOrNull() ?: error("No intent class recognized in input")
 
+    @get:JsonIgnore
+    val sentimets get() = classes.filter { it.type == Class.Type.Sentiment }
```
[the remark this answers is not in the document]
Comment (thePetrMarek) [reply lib/src/main/kotlin/org/promethist/core/Input.kt]: Corrected

File: runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+class Sentiment : Component {
+
+    @Inject
+    lateinit var webTargets: IterableProvider<WebTarget>
+
+    private val webTarget: WebTarget get() = webTargets.named("triton").get()
+
+    private val logger by LoggerDelegate()
+
+    private val idx2Sentiment = mapOf(
```
[the remark this answers is not in the document]
Comment (thePetrMarek) [reply runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt]: Added enum class

File: runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
+package org.promethist.core.nlp
```
[the remark this answers is not in the document]
Comment (thePetrMarek) [reply runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt]: Renamed to Triton

File: lib/src/main/kotlin/org/promethist/core/Input.kt (hunk 1 above)
Comment (thePetrMarek) [reply lib/src/main/kotlin/org/promethist/core/Input.kt]: Error removed, added Class

File: runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt (hunk 2 above)
Comment (thePetrMarek) [reply runner/app/src/main/kotlin/org/promethist/core/nlp/Sentiment.kt]: Try-catch added

**Comment by thePetrMarek** [comment]
I addressed your suggestions for improvements. 
I also noticed, that MutableList classes in Input should contain classes sorted by score, otherwise getter `sentiment` returns negative class only (it was added first into the list). Thus, I added sorting of sentiment classes by score into Triton component.

---TESTING---
I created a testing dialogue in development test/test (Stupid name, I know. I created it in hustle. But, just for explanation, my local admin returns error "Name already exists" for any new name I try). I created four paths through dialogue (POSITIVE, NEUTRAL, NEGATIVE, UNKNOWN). I was able to access all paths. I accessed UNKNOWN path by inserting wrong endpoint URL into app.local.properties file. I also printed content of input.sentiments and input.sentiment by logger. Logger's output is correct.

**Pull request merged**